src/test/REGION_SPECTRAL_PROFILE_STOKES.test.ts

```diff
@@ -0,0 +1,413 @@
+                            resolve();
+                        }
+                    );
+                });
+                await Utility.setEvent(Connection, CARTA.SetCursor, // For getting a SpectralProfileData after changing channel# or stokes#
+                    {
+                        fileId: imageAssertItem.fileId,
+                        point: {x: 0, y:0},
+                        spectralProfiles: [],
+                    }
+                );
+            });
+
+            imageAssertItem.regionGroup.map( (region: Region) => {
+
+                describe(`${region.regionId < 0?"Creating":"Modify"} ${CARTA.RegionType[region.regionType]} region #${region.assert.regionId} on ${JSON.stringify(region.controlPoints)}`, () => {
+                    let SetRegionAckTemp: CARTA.SetRegionAck;
+                    test(`SET_REGION_ACK should return within ${regionTimeout} ms`, async () => {
+                        await Utility.setEvent(Connection, CARTA.SetRegion, region);
+                        await new Promise( resolve => {
+                            Utility.getEvent(Connection, CARTA.SetRegionAck, 
+                                (SetRegionAck: CARTA.SetRegionAck) => {
+                                    SetRegionAckTemp = SetRegionAck;
+                                    resolve();
+                                }
+                            );
+                        });
+                    }, regionTimeout);
+
+                    test("SET_REGION_ACK.success = True", () => {
+                        expect(SetRegionAckTemp.success).toBe(true);
+                    });
+
+                    test(`SET_REGION_ACK.region_id = ${region.assert.regionId}`, () => {
+                        expect(SetRegionAckTemp.regionId).toEqual(region.assert.regionId);
+                    });
+
+                });
+
+                region.profiles.map( profile => {
+                    describe(`${isNumber(profile.stokes)?`Switch to stokes ${profile.stokes} on`:`SET SPECTRAL REQUIREMENTS on ${CARTA.RegionType[region.regionType]}`} region #${region.assert.regionId} with coordinate "${profile.coordinate}"`, () => {
+                        let SpectralProfileDataTemp: CARTA.SpectralProfileData;
+                        test(`SPECTRAL_PROFILE_DATA should return within ${regionTimeout} ms`, async () => {
+                            if (isNumber(profile.stokes)) {
+                                await Utility.setEvent(Connection, CARTA.SetImageChannels, 
+                                    {
+                                        fileId: imageAssertItem.fileId,
+                                        channel: profile.channel,
+                                        stokes: profile.stokes,
+                                    }
+                                );
+                                // await new Promise( resolve => {
+                                //     Connection.onmessage = (messageEvent: MessageEvent) => {
```

Comment:
please keep the code clean without comments

Reply:
Ok, it will be clear after a backend question resolved.